src/Core/Domain/Profile/ValueObject/ProfileId.php

```diff
@@ -59,12 +59,12 @@ public function getValue()
      */
     private function setProfileId($profileId)
     {
-        if (!is_int($profileId) || 0 >= $profileId) {
+        if ((!is_int($profileId) && !ctype_digit($profileId)) || 0 >= $profileId) {
             throw new ProfileException(
                 sprintf('Invalid Profile id %s supplied', var_export($profileId, true))
             );
         }
 
-        $this->profileId = $profileId;
+        $this->profileId = (int) $profileId;
```

Comment:
This goes against strict typing. It's up to the consumer to send the right type.
 
Since this class has already been released with mixed type support and making it stricter would be a BC, we _can_ adapt it but in that case it **must** issue a weak error when receiving a type different than int  (same as we do with deprecations) so that we can change the interface later.

Reply:
@eternoendless I agree sadly we can't add strict typing here because the class was already released in 176. That's why we have so many tests to validate the type... Since we can't validate the strict type we can only validate that the context is a digit
I added the warning logs as you suggested though